collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -729,47 +739,107 @@ protected void onPostExecute(HashMap<String, Object> results) {
+                    dirs.add(new DriveListItem(f.getName(), "", f.getModifiedTime(), "", "",
+                            DriveListItem.DIR, f.getId(), parentId));
+                    break;
+                default:
+                    // skip the rest of the files
+                    break;
+            }
+        }
+        Collections.sort(dirs);
+        Collections.sort(forms);
+        driveList.addAll(dirs);
+        driveList.addAll(forms);
+        checkFormUpdates();
+    }
+
+    private void checkFormUpdates() {
+        for (DriveListItem item: driveList) {
+            if (item.getType() == DriveListItem.FILE) {
+                if (isNewerFormVersionAvailable(item) || areNewerMediaFilesAvailable(item)) {
+                    item.setNewerVersion(true);
                 }
             }
-            Collections.sort(dirs);
-            Collections.sort(forms);
-            driveList.addAll(dirs);
-            driveList.addAll(forms);
-            updateAdapter();
         }
     }
 
+    private boolean isNewerFormVersionAvailable(DriveListItem item) {
+        FormsDao formsDao = new FormsDao();
+        Cursor cursor = formsDao.getFormsCursorForFormFilePath(new StoragePathProvider().getDirPath(StorageSubdirectory.FORMS) + File.separator + item.getName());
+        if (cursor != null && cursor.moveToFirst()) {
+            Long lastModifiedLocal = new File(new StoragePathProvider().getDirPath(StorageSubdirectory.FORMS) + File.separator + item.getName()).lastModified();
+            Long lastModifiedServer = item.getDate().getValue();
+            return lastModifiedServer > lastModifiedLocal;
+        }
+        return false;
+    }
+
+    private boolean areNewerMediaFilesAvailable(DriveListItem item) {
+        String mediaDirName = FileUtils.constructMediaPath(item.getName());
+
+        try {
+            List<com.google.api.services.drive.model.File> mediaFileList;
+            try {
+                mediaFileList = getMediaFiles(item);
+            } catch (MultipleFoldersFoundException exception) {
+                return false;
+            }
+
+            if (mediaFileList != null) {
+                for (com.google.api.services.drive.model.File mediaFile : mediaFileList) {
+                    File localMediaFile = new File(new StoragePathProvider().getDirPath(StorageSubdirectory.FORMS) + File.separator + mediaDirName + File.separator + mediaFile.getName());
+                    if (!localMediaFile.exists()) {
+                        return true;
+                    } else {
+                        Long lastModifiedLocal = localMediaFile.lastModified();
+                        Long lastModifiedServer = mediaFile.getModifiedTime().getValue();
+                        if (lastModifiedServer > lastModifiedLocal) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            Timber.e(e);
```

Comment:
I would log it as a warning not an error so `Timber.w(e);` Since it's not something very important and on the other hand not necessarily our fault.